core/src/trezor/ui/layouts/tt_v2/__init__.py

```diff
@@ -409,35 +403,41 @@ async def show_address(
         if result is CONFIRMED:
             break
 
-        result = await interact(
-            ctx,
-            RustLayout(
-                trezorui2.show_qr(
-                    address=address if address_qr is None else address_qr,
-                    case_sensitive=case_sensitive,
-                    title=title.upper() if title_qr is None else title_qr.upper(),
-                    verb_cancel="XPUBs" if is_multisig else "ADDRESS",
-                )
-            ),
-            "show_qr",
-            ButtonRequestType.Address,
-        )
-        if result is CONFIRMED:
-            break
+        if result is INFO:
```

Comment:
I added some comments in b11c13260cb04c2443ee81fbd359602f70fd23e1, but the loop still sucks. If we were to have more such flows, basically a state machines, it would make sense to describe each with a Rust `enum` + an `impl InteractionFlow` for it describing transitions between the states. But we'd need to teach rust about `ButtonRequests` first so let's not do it here.